.github/workflows/ci-github-actions-self-hosted.yaml

```diff
@@ -214,10 +214,10 @@ jobs:
       matrix:
         jobname:
           [
-            ROCm-Clang13-NoMPI-CUDA2HIP-Real-Mixed,
-            ROCm-Clang13-NoMPI-CUDA2HIP-Real,
-            ROCm-Clang13-NoMPI-CUDA2HIP-Complex-Mixed,
-            ROCm-Clang13-NoMPI-CUDA2HIP-Complex,
+            ROCm-Clang15-NoMPI-CUDA2HIP-Real-Mixed,
+            ROCm-Clang15-NoMPI-CUDA2HIP-Real,
+            ROCm-Clang15-NoMPI-CUDA2HIP-Complex-Mixed,
+            ROCm-Clang15-NoMPI-CUDA2HIP-Complex,
```

Comment:
I'd like to move all the job names prefix to
```
HARDWARE-COMPILER(VERSION)-FEATURES
```
HARDWARE: when using GPU, mention GPU, otherwise mention CPU.
COMPILER. ROCm compiler should be versioned as ROCm instead of clang. Version can be optional.
So
Radeon7-ROCm-NoMPI-CUDA2HIP-Real-Mixed.

Reply:
Q. Is LLVM 15.0.7 reliable enough to use in CI with AMD? (We'll find out...)

Please also add the amdclang compiler that is already installed, either here or in a later PR.

Reply:
Also: Lets try NOT reinstating any CUDA builds on nitrogen.

Reply:
@ye-luo done in the new commit. @prckent amdclang is shipped inside rocm. We are keeping `ROCM_PATH` as default from `/opt/rocm` ideally the latest possible version.

Reply:
> Also: Lets try NOT reinstating any CUDA builds on nitrogen.

So CUDA-AFQMC-* should be gone?

Reply:
My bad. Could you chang Radeon7 to RadeonVII?

Reply:
@prckent llvm 15 and the upcoming 16 are not ready on amdgpu.

Reply:
@williamfgc so for the offload builds we use amdclang only. These can be added in a later PR.

Reply:
And my understanding is that the only reason CUDA2HIP is enabled here is to test that the AFQMC code works. Hopefully this is enabled and runs correctly.  (No time to look for several days)

Reply:
AFQMC HIP have copied source code with suffix `.hip.cpp` (ENABLE_HIP). We are not testing them in CI.

Reply:
On sulfer we test
clang + offload + cuda + afqmc, named Clang15-MPI-CUDA-AFQMC-Offload
On nitrogen
gcc + cuda +afqmc named GCC9-MPI-CUDA-AFQMC

I think we need to running GCC9-MPI-CUDA-AFQMC before revising the CI matrix.